### ISSUE TITLE
A tree builder without a root node is deprecated since Symfony 4.2

### ISSUE THREAD

**Issue by belinde** [doctrine/DoctrineMigrationsBundle#251]
In `DependencyInjection\Configuration::getConfigTreeBuilder()` a TreeBuilder is istantiated without arguments, and so triggers an `E_USER_DEPRECATED` error on Symfony 4.2. The problem is in the first two rows of method:
```php
        $treeBuilder = new TreeBuilder();
        $rootNode    = $treeBuilder->root('doctrine_migrations', 'array');
```
I think it could be safe put the parameters ALSO in the constructor, so the error is suppressed in Symfony 4.2 but the code still works for previous versions:
```php
        $treeBuilder = new TreeBuilder('doctrine_migrations', 'array');
        if (!is_callable([$treeBuilder, 'getRootNode'])) { // hack for backward compatibility
            $rootNode = $treeBuilder->root('doctrine_migrations', 'array');
        }
```
In this way new versions get the arguments in the constructor, and old versions silently ignore them. I assume that `getRootNode()` has been introduced in the same time of the deprecation, so it could be a good and almost inexpensive check.

**Comment by greg0ire** [comment]
Can you make a PR? Here is an example: https://github.com/sonata-project/SonataAdminBundle/pull/5360

**Comment by alcaeus** [comment]
Yes, a PR would be great. Please make sure to send it to the 1.3 branch as we want to fix the error not only in 2.0 (which is master) but 1.3 as well. 👍

**Comment by belinde** [comment]
I wasn't sure about the quality of the fix... Sure, I'll do tomorrow, I don't have a PC available right now.

**Comment by belinde** [comment]
In branch 1.3 the issue has been fixed in commit e16ce0c78a9f84b67cbd1a6c846414e26a32e4f4 by user Chalas, in date 2018-12-01, so it has just to be merged in master.

**Comment by jwage** [comment]
This was merged to master and is in 2.0
